### PR TITLE
Setting up RSpec Testing 

### PR DESCRIPTION
**Pull request by chelobotix**
In this PR we fulfilled the following tasks:
- [x] Add unit tests for all the methods in all the classes:

- Book
- Person
- Student
- Teacher
- Classroom
- Rental
- Decorator Classes

### CODE REVIEW

File: spec/person_spec.rb [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+      expect(@person.can_use_services?).to eql true
+    end
+
+    it 'Rented Book should be added to book rental list' do
+      @person.add_rental(@rental)
+      puts @book.rentals
+      expect(@book.rentals.length).to eql 1
+    end
+  end
+end
```
Comment (VICTOR-CHUKWU) [inline spec/person_spec.rb]: - [ ] good job, but some tests are missing, you are missing tests to check if `The constructor method sets default value for the optional name parameter if it is not provided` and also tests for the method `correct_name`

File: spec/student_spec.rb [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+describe Student do
+  before :each do
+    @student = Student.new 24, 'Eddie'
+  end
+  context 'When testing the Student class' do
+    it 'Given age and a name Object must be returned' do
+      expect(@student).to be_an_instance_of Student
+    end
+  end
```
Comment (VICTOR-CHUKWU) [inline spec/student_spec.rb]: - [ ] Good job, but some tests are missing, you are missing tests to check if `The constructor method sets the default value for the optional name parameter if it is not provided `

File: spec/teacher_spec.rb [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+describe Teacher do
+  before :each do
+    @teacher = Teacher.new 45, 'Biology'
+  end
+  context 'When testing the Teacher class' do
+    it 'Given age, and specialization teacher object must be returned' do
+      expect(@teacher).to be_an_instance_of Teacher
+    end
+  end
+end
```
Comment (VICTOR-CHUKWU) [inline spec/teacher_spec.rb]: - [ ] Good job, but some tests are missing, you are missing tests to check if `The constructor method sets the default value for the optional name parameter if it is not provided and also tests for the method `can_use_services?`

**Review by VICTOR-CHUKWU — CHANGES_REQUESTED** [review]
Hi @gealsanchez  👋  

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

almost

## Highlights
- Good PR
- Good commit
- All linters passed


## Required Changes ♻️

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by amejid — APPROVED** [review]
Hi @gealsanchez & @chelobotix, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights
- Linter checks passing :heavy_check_mark: 
- Correct use of Gitflow :heavy_check_mark: 
- Methods covered in tests :heavy_check_mark: 
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**